src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js

```diff
@@ -10,11 +12,60 @@ type Props = {
     skin?: Skin,
 };
 
-export default class Items extends React.PureComponent<Props> {
+@observer
+export default class Items extends React.Component<Props> {
+    @observable expandedWidth = 0;
+    @observable showText = true;
+
     static defaultProps = {
         skin: 'light',
     };
 
+    parentRef: { current: null | HTMLDivElement } = React.createRef();
+    childRef: { current: null | HTMLUListElement } = React.createRef();
+
+    componentDidMount() {
+        this.setExpandedWidth();
+        window.addEventListener('resize', this.setExpandedWidth);
+    }
+
+    componentDidUpdate() {
+        this.setExpandedWidth();
+    }
+
+    @action setExpandedWidth = () => {
```

Comment:
If we agree on below comment about a computed property we could make this a `setDimensions` function, which is called when something is resized, and `this.setShowText` wouldn't have to be called in here.